src/schema.ts

```diff
@@ -0,0 +1,45 @@
+import { gql } from "apollo-server";
+
+export const typeDefs = gql`
+  type Customer {
+    id: String!
+    name: String!
+    email: String!
+  }
+
+  type LendingRecord {
+    customer: Customer!
+    lentDate: String!
+    returnDate: String
+    dueDate: String!
+  }
+
+  type Book {
+    id: String!
+    title: String!
+    author: String!
+    isLent: Boolean!
+    currentLendee: Customer
+    dueDate: String
+    lendingHistory: [LendingRecord!]!
```

Comment:
I guess those can be marked as optional, as not every Book will be lent at the time of its creation. 

Also, do you think `dueDate` should be saved for the Book entry, or for LendingRecord entry? I think the Book itself does not have a dueDate(except if it is not food and has expiry date), but more like the dueDate should be filled when a Customer takes a Book and we create a LendingRecord that connects Book to Customer

Reply:
I agree with you, I will remove the dueDate here and leave it only in the lendingRecord.